src/main/java/uk/gov/hmcts/darts/arm/client/model/rpo/MasterIndexFieldByRecordClassSchemaResponse.java

```diff
@@ -1,13 +1,99 @@
 package uk.gov.hmcts.darts.arm.client.model.rpo;
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import lombok.Data;
 import lombok.EqualsAndHashCode;
 import lombok.NoArgsConstructor;
 
+import java.util.List;
+
 @JsonIgnoreProperties(ignoreUnknown = true)
 @Data
 @NoArgsConstructor
 @EqualsAndHashCode(callSuper = true)
 public class MasterIndexFieldByRecordClassSchemaResponse extends AbstractMatterResponse {
+
+    private List<MasterIndexField> masterIndexFields;
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    @Data
+    @NoArgsConstructor
+    @EqualsAndHashCode
+    public static class MasterIndexField {
```

Comment:
I don't agree. I think we should get everything that can be sent so we can see whether what they say they are sending is actually what we get. This way if they change things as ARM are known for doing, we can see what changes are made.

Reply:
Ok but one thing to consider - the more response fields we attempt to deserialise the more opportunity there is for deserialisation to fail (e.g. if ARM were to change a datatype). For that reason, it's usually safer to minimise exposure by deserialising only what our application needs to do its job.

If we attempt deserialisation for fields we're not using then it introduces risk, the upside of which isn't clear to me. The downside is clearer - changes made by ARM to those unused fields could break our application in production, even although we don't otherwise care about that data.

If the goal is just to log the full request/response, then configuring logging on the Feign client is an option that doesn’t introduce any application risk (https://www.baeldung.com/java-feign-logging).

Reply:
If we deserialise and it fails because they changed the format, surely we need to know that?

Reply:
Spoke to Hemanta and he said to just use the values we need as those are what is agreed upon with ARM and they are likely to change the other values. So if thats the case should the stubs also be changed so they only return the expected fields as they can then get out of sink with what ARM return and this will not be easily identified?

Reply:
> If we deserialise and it fails because they changed the format, surely we need to know that?

Yep, the outcome will be that a `DecodeException` is thrown and the thread is killed, so in that sense we'll know. This is an extension of `FeignException`, so our try/catch will handle it and we'll get visibility just like with any other failure. This is the point - this scenario would break the functionality, so we want to minimise the chances of that happening by minimising what we decode.

Suggest we keep the stub as-is, as it's a realistic representation of what ARM's API provides - there's still value in demonstrating that our deserialisation can handle a subset of the raw response (which should be the case as we have `@JsonIgnoreProperties(ignoreUnknown = true)` in our response classes. 

I take your point on how do we know if the unused fields in the stubbed data gets out step with ARM, but I don't think that's a problem we should try to solve here in the application code.